lib/datadog/open_feature/exposures/buffer.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require_relative '../../core/buffer/cruby'
+
+module Datadog
+  module OpenFeature
+    module Exposures
+      # This class is a buffer for exposure events that evicts at random and
+      # keeps track of the number of dropped events
+      class Buffer < Core::Buffer::CRuby
+        DEFAULT_LIMIT = 1_000
+
+        def initialize(limit = DEFAULT_LIMIT)
+          @dropped = 0
+
+          super
+        end
+
+        protected
+
+        def drain!
+          drained = super
+
+          dropped = @dropped
+          @dropped = 0
+
+          [drained, dropped]
+        end
```

Comment:
So... I'm not sure returning `dropped` here is a good idea. Specifically, we're breaking the "L" in [SOLID](https://en.wikipedia.org/wiki/SOLID) -- we now have a method that doesn't conform to the contract of the superclass (it's a completely different type).

The problem with doing this is that if the superclass changes to e.g. do something with the result of calling `drain!` (let's say it implements a helper that does `drain!.map`) then we'll be breaking that.

TL;DR my suggestion is to keep the parent class's return AND separately either add an accessor for `dropped` or add a `drain_with_count` (or some other name) that returns the new type. This way we'll never break expectations from the parent class.

Reply:
Good idea 👍🏼